SofaKernel/modules/SofaBaseCollision/SphereModel.h

```diff
@@ -181,8 +198,10 @@ class SphereCollisionModel : public core::CollisionModel
 
     void computeBBox(const core::ExecParams* params, bool onlyVisible=false) override;
 
+    SphereActiver *myActiver; ///< SphereActiver that activate or deactivate collision sphere during execution
 protected:
     core::behavior::MechanicalState<DataTypes>* mstate;
+    Data<std::string> SphereActiverPath; ///< path of a component SphereActiver that activate or deactivate collision sphere during execution
```

Comment:
instead of Data<std::string>, use Link for component path and stuff

Reply:
Again, I applied the same pattern as it was coded for Point/Line activers.
You probably mean I should correct Sphere Line and Point activers then ? Would not it break some plugin controller code ? Do we care about it ?

Reply:
You are right about breaking things, but it is for the sake of cleaning 🤽‍♂ (just add breaking flag)
Personally I would bring at least this part (Link) to the other line/point activers ; especially if we suppose that the task #1262 wont be completed before a while...
Just my cent though 🦆

Reply:
To me, if a feature is duplicated in more than two component...is a strong indication that it is time to re-factor it. I may be wrong but do #1262 task so complex ?